datafusion/core/src/physical_plan/file_format/file_stream.rs

```diff
@@ -98,6 +99,8 @@ enum FileStreamState {
         partition_values: Vec<ScalarValue>,
         /// The reader instance
         reader: BoxStream<'static, Result<RecordBatch, ArrowError>>,
+        /// A [`FileOpenFuture`] for the next file to be processed
+        next: Option<(FileOpenFuture, Vec<ScalarValue>)>,
```

Comment:
I wonder if we could make it future-proof by potentially prefetching `n` files instead of 1? I guess in cases where file opening is slower than scanning / processing, this could make a difference (e.g. small files).

Reply:
Perhaps a follow on PR could turn this into a stream and use `StreamExt::buffered` or something

Reply:
yeah, that seems like a good idea